src/pincode/PincodeEnter.tsx

```diff
@@ -1,126 +1,106 @@
 /**
- * This is a reactnavigation SCREEN to which we navigate,
+ * This is a react navigation SCREEN to which we navigate,
  * when we need to fetch a PIN from a user.
  */
 import { StackScreenProps } from '@react-navigation/stack'
-import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
+import React, { useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
 import { StyleSheet } from 'react-native'
 import { SafeAreaView } from 'react-native-safe-area-context'
-import { connect } from 'react-redux'
 import { AuthenticationEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
 import { ErrorMessages } from 'src/app/ErrorMessages'
-import { withTranslation } from 'src/i18n'
 import { headerWithBackButton } from 'src/navigator/Headers'
 import { modalScreenOptions } from 'src/navigator/Navigator'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 import { checkPin } from 'src/pincode/authentication'
 import Pincode from 'src/pincode/Pincode'
-import { RootState } from 'src/redux/reducers'
+import useSelector from 'src/redux/useSelector'
 import { currentAccountSelector } from 'src/web3/selectors'
 
-interface State {
-  pin: string
-  errorText: string | undefined
-  pinIsCorrect: boolean
-}
-
-interface StateProps {
-  currentAccount: string | null
-}
-
-type RouteProps = StackScreenProps<StackParamList, Screens.PincodeEnter>
-type Props = StateProps & WithTranslation & RouteProps
+type Props = StackScreenProps<StackParamList, Screens.PincodeEnter>
 
-class PincodeEnter extends React.Component<Props, State> {
-  static navigationOptions = () => ({
-    ...modalScreenOptions(),
-    ...headerWithBackButton,
-    gestureEnabled: false,
-  })
+export const PincodeEnter = ({ route }: Props) => {
+  const { t } = useTranslation()
+  const [pin, setPin] = useState('')
+  const [errorText, setErrorText] = useState(undefined)
+  const [pinIsCorrect, setPinIsCorrect] = useState(false)
+  const currentAccount = useSelector(currentAccountSelector)
+  const pinRef = useRef<string>('')
```

Comment:
why do we need this `pinRef`? seems like we're just trying to keep it in sync with `pin` so not sure what this is for?

Reply:
> @kathaypacific without the ref I wasn't getting the most up to date pin value from state so the pin passed to `onPressConfirm` and was sometimes 1 digit short.

@MuckT ah i see what you mean. 

i think the issue is actually in the `Pincode.tsx` screen, particularly in this section: https://github.com/valora-inc/wallet/blob/36fc7d05813032785bd14815bc7dbdb11025731e/src/pincode/Pincode.tsx#L38-L42 - i think that we're trying to use `requestAnimationFrame` as a "hack", to make it so that the previous state change (the `onChangePin` state change) happens before executing the `onCompletePin`. i say this is "hack" because it's a little brittle, and relies on the state change actually happening during the animation frame - which in this case after the refactor, seems to be not true anymore.

i think a more correct way of achieving this is actually waiting for the value of `pin` to be set, then doing the `onCompletePin` - since we're waiting to do something as an effect of a state change, we could remove that code block i liked above, and instead add something like ```
  useEffect(() => {
    if (pin.length === maxLength) {
      onCompletePin(pin)
    }
  }, [pin])
``` 

i *think* that if this is done in the Pincode.tsx file, then you can remove the pincode ref in the PincodeEnter.tsx file and be fine with just the `pin` in the state :) happy to discuss in person if this is not clear!

Reply:
Fixed in 5e4ea9fb550d0d31114d3918bd1ee4a6a77be679!